faucet/config_parser_util.py

```diff
@@ -98,31 +101,34 @@ def dp_include(config_hashes, config_file, logname, top_confs):
         new_top_confs[conf_name] = curr_conf.copy()
         try:
             new_top_confs[conf_name].update(conf.pop(conf_name, {}))
-        except TypeError:
+        except (TypeError, ValueError):
             logger.error('Invalid config for "%s"' % conf_name)
             return False
 
-    for include_directive, file_required in (
-            ('include', True),
-            ('include-optional', False)):
-        for include_file in conf.pop(include_directive, []):
-            if not isinstance(include_file, str):
-                include_file = str(include_file)
-
-            include_path = dp_config_path(include_file, parent_file=config_file)
-            if include_path in config_hashes:
-                logger.error(
-                    'include file %s already loaded, include loop found in file: %s',
-                    include_path, config_file,)
-                return False
-            if not dp_include(
-                    new_config_hashes, include_path, logname, new_top_confs):
-                if file_required:
-                    logger.error('unable to load required include file: %s' % include_path)
+    try:
+        for include_directive, file_required in (
+                ('include', True),
+                ('include-optional', False)):
+            for include_file in conf.pop(include_directive, []):
+                if not isinstance(include_file, str):
+                    include_file = str(include_file)
+
+                include_path = dp_config_path(include_file, parent_file=config_file)
+                if include_path in config_hashes:
+                    logger.error(
+                        'include file %s already loaded, include loop found in file: %s',
+                        include_path, config_file,)
                     return False
-                else:
-                    new_config_hashes[include_path] = None
-                    logger.warning('skipping optional include file: %s' % include_path)
+                if not dp_include(
+                        new_config_hashes, include_path, logname, new_top_confs):
+                    if file_required:
+                        logger.error('unable to load required include file: %s' % include_path)
+                        return False
+                    else:
+                        new_config_hashes[include_path] = None
+                        logger.warning('skipping optional include file: %s' % include_path)
+    except TypeError:
+        assert False, 'values in include directive are invalid'
```

Comment:
I wonder if it would be useful to be able to print which included file had the error in? Or will we get this printed regardless because we will have an earlier exception that will fire?

Reply:
This error is thrown when the included value is not in a proper format eg:
```
include: 'acls.yaml'
vlans:
    office:
        vid: 100
dps:
    sw1:
        dp_id: 0x1
        interfaces:
            1:
                native_vlan: office
```
instead of
```
include: 
    - acls.yaml
vlans:
    office:
        vid: 100
dps:
    sw1:
        dp_id: 0x1
        interfaces:
            1:
                native_vlan: office
```
I have uploaded some changes that deals with this better.